docs/UsingSeal.md

```diff
@@ -214,55 +221,59 @@ To decrypt an encrypted object in a Move package, follow these steps:
   ],
 });
+const allPublicKeys = [publicKey]; // can be many
 ```
 
-Assume you have:
+Now lient can first verify the keys, then decrypt using the Move functions. 
 
-- `encryptedBytes`: a BCS-serialized encrypted object,
-- `txBytes`: a transaction block that calls a `seal_approve*` function (see [Decryption](#decryption)).
-- `allPublicKeys`: an array of Move objects for all public keys in the encryption,
-- `correspondingPublicKeys`: the public keys that correspond to the derived keys.
+Client should have:
+
+- `encryptedBytes`: the BCS-serialized encrypted object.
+- `txBytes`: a valid transaction block that satisfies the `seal_approve*` policy function (see [Decryption](#decryption)).
+- `sessionKey`: a session key for the given package (see [Decryption](#decryption)).
+- `allPublicKeys`: an array of Move objects for all public keys in the encryption.
+- `correspondingPublicKeys`: the public keys that correspond to the derived keys. // whats this difference btw allPublicKeys vs correspondingPublicKeys????
 
-A transaction for on-chain decryption could look like this:
 ```typescript
-// Parse BCS serialized encrypted object
+// Parse BCS serialized encrypted object from bytes.
 const encryptedObject = EncryptedObject.parse(encryptedBytes);
 
-// Get derived keys from key servers
-const derivedKeys = await sealClient.getDerivedKeys({
-  id: encryptedObject.id,
-  txBytes, // Should contain call to seal_approve as discussed in the [decryption](#Decryption) section
-  sessionKey,
-  threshold: encryptedObject.threshold,
-});
-
-// Parse encrypted object as Move object
+// Alternatively, parse encrypted object as Move object with `parse_encrypted_object`.
 // For some applications, this object should be stored on-chain (e.g., encrypted votes should be stored before the deadline, and decrypted after).
 const tx = new Transaction();
-const parsedEncryptedObject = tx.moveCall({
+const encryptedObject = tx.moveCall({
   target: `${SEAL_PACKAGE_ID}::bf_hmac_encryption::parse_encrypted_object`,
-  arguments: [tx.pure.vector("u8", Array.from(encryptedObject.))],
+  arguments: [tx.pure.vector("u8", Array.from(encryptedBytes))],
 });
 
+// Get derived keys from key servers.
+const derivedKeys = await sealClient.getDerivedKeys({
+  id: encryptedObject.id,
+  txBytes,
+  sessionKey,
+  threshold: encryptedObject.threshold,
+});
 
-// Convert the derived keys to G1 elements
+// Convert the derived keys to G1 elements.
 const derivedKeysAsG1Elements = Array.from(derivedKeys).map(([derivedKey]) => 
   tx.moveCall({
     target: `0x2::bls12381::g1_from_bytes`,
     arguments: [tx.pure.vector("u8", fromHex(derivedKey.toString()))],
   }),
 );
 
-// Verify the derived keys. This can be cached if decryption for the same ID is done again
+// Call to verify the derived keys. This can be cached if decryption for the same ID is done again. 
+// what does this mean `for the same ID is done again.`????
```

Comment:
```suggestion
```

Reply:
not necessarily the same encrypted object: but for the same package id + id